src/core/type.jl

```diff
@@ -9,11 +9,9 @@ identify(::Type{LLVMType}, ::Val{K}) where {K} = bug("Unknown type kind $K")
 
 @inline function check(::Type{T}, ref::API.LLVMTypeRef) where T<:LLVMType
     ref==C_NULL && throw(NullException())
-    @static if DEBUG
+    @debug begin
```

Comment:
This seems like an odd use of `@debug` to contain an `@assert`.

Really `@debug` is for emitting verbose, low priority messages which may be ignored. The expression passed to `@debug` should evaluate to a human-readable message.

Reply:
Hmm, I wanted to avoid (DRY) another global `DEBUG` setting that determines whether costly checks are enabled, and (ab)using `@debug` seemed like a prime candidate for that. I see now how it still prints something.